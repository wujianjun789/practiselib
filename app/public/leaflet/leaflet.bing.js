/**
 * Created by a on 2018/1/9.
 */
