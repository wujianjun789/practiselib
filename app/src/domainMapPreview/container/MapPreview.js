/**
 * Created by a on 2017/7/25.
 */
import React,{Component} from 'react'

import '../../../public/styles/domain-mapPreview.less';
import Content from '../../components/Content'
import MapView from '../../components/MapView'

import SearchText from '../../components/SearchText'

import {getDomainList} from '../../api/domain'

export default class MapPreview extends Component{
    constructor(props){
        super(props)
        this.state = {
            mapId:"mapPreview",
            domainList:[{id:1, name:'domain1', "device_type": 'DEVICE', geoPoint:{lat:31.239658843127756, lng:121.49971691534425}},
                {id:2, name:'domain2', "device_type": 'DEVICE', geoPoint:{lat:31.242658843127756, lng:121.49972691534425}}],
            search:{placeholder:'输入域名称搜索', value:'', curIndex:0},

            curDomain: 0,
            panLatlng:null
        }

        this.onChange = this.onChange.bind(this);
        this.itemClick = this.itemClick.bind(this);
        this.searchSubmit = this.searchSubmit.bind(this);
        this.panCallFun = this.panCallFun.bind(this);
        this.initDomainList = this.initDomainList.bind(this);
    }

    componentWillMount(){
        this.mounted = true;
        // getDomainList(data=>{ this.mounted && this.initDomainList(data)})
    }

    componentWillUnmount(){
        this.mounted = false;
    }

    initDomainList(data){
        this.setState({domainList:data});
    }

    onChange(value){
        const { search } = this.state;

        let newValue = Object.assign({}, search, {value:value});
        this.setState({search:newValue});
    }

    itemClick(itemIndex){
        const {search} = this.state
        let newValue = Object.assign({}, search, {curIndex:itemIndex});
        this.setState({search:newValue});
    }

    searchSubmit(){
        const {domainList, search} = this.state;
        for(var key in domainList){
            let item = domainList[key];
            if(!search.value || item.name.indexOf(search.value)>-1){
                this.setState({panLatlng:item.geoPoint});
                break;
            }
        }
    }

    panCallFun(){
        console.log('callFun');
        this.setState({panLatlng:null});
    }

    render(){
        const {mapId, domainList, search, curDomain, panLatlng} = this.state;
        let positionList = domainList.map(item=>{
            let geoPoint = item.geoPoint ? item.geoPoint : {lat:"", lng:""};
            return Object.assign(geoPoint, {"device_type":"DEVICE"}, {"device_id":item.id});
        })

        let datalist = [];
        for(var key in domainList){
            let item = domainList[key];
            if(!search.value || item.name.indexOf(search.value)>-1){
                datalist.push({id:item.id, value:item.name})
            }
        }

        let domain = domainList && domainList.length  ? domainList[curDomain]:null;
        let latlng = domain && domain.geoPoint ? domain.geoPoint : null;
        return <Content className="map-preview">
            <MapView mapData={{id:mapId, latlng:latlng, position:positionList, data:domainList}}
                     panLatlng={panLatlng} panCallFun={this.panCallFun}/>
            <SearchText IsTip={true} datalist={datalist} placeholder={search.placeholder} value={search.value}
                        onChange={this.onChange} itemClick={this.itemClick} submit={this.searchSubmit}/>
            </Content>
    }
}