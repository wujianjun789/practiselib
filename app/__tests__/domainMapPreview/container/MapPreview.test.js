/**
 * Created by a on 2017/7/27.
 */
import React from 'react';
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'

import '../../../public/js/jquery-3.1.1.min'
import '../../../public/leaflet/leaflet';
window.d3 = require('../../../public/js/d3.min')

import MapPreview from '../../../src/domainMapPreview/container/MapPreview';


test('MapPreview renders', ()=>{
    const component = renderer.create(
        <MapPreview />
    )

    let mapPreview = component.toJSON();
    expect(mapPreview).toMatchSnapshot();
})

test('MapPreview div click', ()=>{
    const component = shallow(<MapPreview />)
})