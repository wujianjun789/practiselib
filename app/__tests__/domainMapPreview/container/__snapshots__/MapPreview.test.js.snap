exports[`test MapPreview renders 1`] = `
<div
  className="content map-preview">
  <div
    className="map-view "
    id="mapPreview" />
  <div
    className="searchText "
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}>
    <input
      onChange={[Function]}
      placeholder="输入域名称搜索"
      type="search"
      value="" />
    <ul
      className="">
      <li
        className=""
        onClick={[Function]}
        value="domain1">
        domain1
      </li>
      <li
        className=""
        onClick={[Function]}
        value="domain2">
        domain2
      </li>
    </ul>
    <span
      className="glyphicon glyphicon-search"
      onClick={[Function]} />
  </div>
</div>
`;
